src/lib.rs

```diff
@@ -6,9 +6,11 @@ use {
     std::{borrow::Cow, collections::HashMap},
 };
 
+type HandlerMap<T, A, E> = HashMap<&'static str, Box<dyn FnMut(&mut T, &mut A, E)>>;
```

Comment:
Since the type was changed from `Fn` to `FnMut`, you also need to update `on` and `and_on` parameters to reflect this.